CMakeLists_files.cmake

```diff
@@ -78,6 +85,21 @@ list( APPEND PUBLIC_HEADER_FILES
       opm/common/util/numeric/cmp.hpp
       opm/common/utility/platform_dependent/disable_warnings.h
       opm/common/utility/platform_dependent/reenable_warnings.h
+      opm/core/utility/Average.hpp
+      opm/core/utility/compressedToCartesian.hpp
+      opm/core/utility/DataMap.hpp
+      opm/core/utility/Event.hpp
+      opm/core/utility/Event_impl.hpp
+      opm/core/utility/Factory.hpp
+      opm/core/utility/have_boost_redef.hpp
+      opm/core/utility/linearInterpolation.hpp
+      opm/core/utility/MonotCubicInterpolator.hpp
+      opm/core/utility/NullStream.hpp
+      opm/core/utility/RegionMapping.hpp
+      opm/core/utility/RootFinders.hpp
+      opm/core/utility/share_obj.hpp
+      opm/core/utility/SparseVector.hpp
+      opm/core/utility/UniformTableLinear.hpp
```

Comment:
did you check that all of these files are needed? e.g., `UniformTableLinear.hpp`, `share_obj.hpp` and `has_boost_redef.hpp` look suspicious (plus most of the other files)...

Reply:
no i did not check the usage for the templates. but those with cpp files are definitely needed. will do i guess.

Reply:
I think that we should get rid of as much stuff as possible as long as we have the opportunity: if the only user of some piece of code is the legacy flow code and/or opm-upscaling, the class should move to opm-simulators. probably opm-upscaling will temporarily need to depend on opm-simulators, but I don't think that this is much of a problem.

Reply:
likewise for `opm-grid`: `SparseVector` seems to belong there rather than opm-common.

Reply:
yes. i didn't disagree as such. however, i need instructions from those in control of the process and those were to move utilities to opm-common. anyways,

- Average.hpp can apparently be moved to upscaling and is thus left out of this pr.
- compressedToCartesian.h/cpp is only used in opm-core itself so it is left there for the time being.
- DataMap.hpp can be moved to simulators and thus left out of this pr.
- linearInterpolation.hpp can be moved to simulators and thus left out of this pr.
- MonotCubicInterpolator.hpp is used in core and upscaling. left out of this pr.
- NullStream.hpp can be moved to simulators and thus left out of this pr.
- Factory.hpp is unused
- RegionMapping is used in core and simulators. left out until core code has been settled
- share_obj.hpp can be moved to simulators
- SparseVector.hpp is used in simulators and upscaling. moved to common.
- UniformTableLinear.hpp is only used in core.
- find_boost_redef.hpp is unused.

i think having simulators as an upstream in upscaling is a no go. it makes little sense to depend on something as heavy for some utility classes only.

anyways, this butchered this PR to only move SparseVector and associated test. @atgeirr should we move this to common or grid as suggested ? (to me having a generic Sparse vector class in grid sounds weird but..)

Reply:
> to me having a generic Sparse vector class in grid sounds weird but..

I agree here. Not so sure if we should get rid of it altogether, though.